what-is-chalk.mdx

```diff
@@ -18,215 +18,440 @@ import { PyDiffEditor } from '@/components/Editor'
-sources -- join data from services like S3, Redshift, BQ, Snowflake
-or other custom sources with historical features computed online.
-Specify a cron schedule on an `@offline` resolver, and Chalk will automatically ingest
-data with support for incremental reads:
-
-```sql
--- resolves: User
--- source: snowflake
--- cron: 1h
--- incremental:
---   lookback_period: 10m
---   incremental_column: ts
-select
-    id,
-    credit_score as start_of_day_balance,
-    ts
-from credit_scores
-```
+We wrote an [engineering blog post on our Symbolic Python interpreter](https://chalk.ai/blog/symbolic-python-interpreter) if you want to learn more!
 
-Chalk makes this data available for point-in-time-correct dataset
-generation for data science use cases. Every pipeline has built-in
-monitoring and alerting to ensure data quality and timeliness.
+### How can I cache features in Chalk?
 
-### Reverse ETL
+Some data sources–like LLMs or 3rd party APIs are expensive or slow to call at inference time,  Chalk supports caching features to optimize for latency and cost:
 
-When your model needs to use features that are canonically stored in
-a high-latency data source (like a data warehouse), Chalk's Reverse
-ETL support makes it simple to bring those features online and serve
-them quickly.
+- define cache windows inline with your features
+- override staleness at query-time when fresh data is critical
+- pre-warm caches and backfill pipelines for performance
 
-Add a single line of code to an `offline` resolver, and Chalk constructs
-a managed reverse ETL pipeline for that data source:
+Add a caching policy with one line of code in our feature definition:
 
-```py
-@offline(offline_to_online_etl="5m")
-```
+<PyDiffEditor html={addCachingCode} />
 
-Now data from slow offline data sources is automatically available for
-low-latency online serving.
 
----
+### How do I deploy and query features with Chalk?
+
+Once we've defined our pipelines, deploy features to a branch in < 30s with:
 
-## Deploying and querying
+```shell
+chalk apply --branch=elvis
+```
+
+We can then query for features against this branch and perform QA workflows on our preview environment to make sure that our Chalk deployment performs as expected:
+
+```shell
+chalk query --branch=elvis --in user.id=47
+```
 
-Once you've defined your pipelines, you can rapidly deploy them to
-production with [Chalk's CLI](/cli/introduction):
+Once validated, promote the deployment to production,
+which is served at a unique preview URL.
 
-```bash
+```shell
 chalk apply
 ```
 
-This creates a deployment of your project, which is served at a unique
-preview URL. You can promote this deployment to production, or
-perform QA workflows on your preview environment to make sure that
-your Chalk deployment performs as expected.
+These features are now available for low-latency online inference and [offline training](https://docs.chalk.ai/docs/training-client).
```

Comment:
can we link online inference here?

Reply:
hyperlinked to https://docs.chalk.ai/docs/query-basics